### DISCUSSION TITLE
Free Effect Plugins powered by Camomile!

### DISCUSSION THREAD

**Discussion by fbarton112** [pierreguillot/Camomile#317] [Show and tell]
Hey everyone! 

Camomile has been an absolute game changer for me. Over the last couple years, i've been working on a collection of audio effect plugins for music production, and I want to share them with the community here! 

You can download them here: [monkwind.gumroad.com](https://monkwind.gumroad.com/)

Thank you so much Pierre for this amazing project!! merci bcp

![DrumWizard-gif2](https://user-images.githubusercontent.com/60185387/226063159-2894c91d-ab36-4baf-a160-fc0f363a0694.gif)

![MonkwindEffectBundle](https://user-images.githubusercontent.com/60185387/226061798-51abfd05-bd2c-4c4a-8ed2-0e2fb97f589a.jpg)

**Comment by piedagile** [comment]
Hello fbarton112

Two tears ago I also developed an Apple effect with Camomile, with the idea of a graphical pad for controlling parameters. 

However, Pierre Guillot pointed out that it wasn't possible to incorporate Pure Data codes for the graphics in Camomile.  
So I got round this by using an external PD xy drag graphic that communicates with Logic via MIDI codes.

I see that you've managed to put graphics in your modules, is it possible for you to provide some explanations, or even advice. 

Thanks in advance, Charles

**Reply by fbarton112** [reply]
Hey Charles,

I am exclusively using basic Pure Data UI objects in these plugins - nothing but Sliders, Toggles, Bangs, and Arrays. Each plugin also has a background image i created using photoshop. There are a couple instances (GimmeSpace and MajicFM) where the user can draw a custom curve for certain parameters. This is done with an Array object, the contents of which are then saved in the plugin state ([here's documentation about saving arrays](https://github.com/pierreguillot/Camomile/wiki/How-to-create-new-plugins#programs))

I'm happy to chat if you have any questions!
-Fletcher

**Comment by BrebisSauce21** [comment]
Hello, cool plugins ! Just downloaded the saturator and it sounds great
I don't know how to display the arrays in the plugin like you, could you explain it to me ?

**Reply by BrebisSauce21** [reply]
It seems that with tabwrite you can display the waveform on the array but not with tabwrite~

**Reply by fbarton112** [reply]
https://github.com/pierreguillot/Camomile/assets/60185387/93a1cb79-f101-4ae3-ab28-a53556c47d18

Hey thanks! glad you like em. This is the basic method i use whenever i need to display dynamic amplitude data on an array. In this case, my array has a size of 100 and a Y range from 0 to 1.

**Reply by BrebisSauce21** [reply]
Thanks for your answer, it looks quite handy for effect plugins. In my case it's to put my waveshaper enveloppe in the vst, but I don't think signals in an array are supported in camomille (visually).

https://github.com/pierreguillot/Camomile/assets/77810320/3b07ce48-5eb4-4a50-b9ce-68cbc431b682

**Reply by BrebisSauce21** [reply]
Sorry for my messy patch btw, i'm a beginner in pure data
